simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -818,5 +818,93 @@ auto EntityBase::updateTraveledDistance(const double step_time) -> double
   return traveled_distance_;
 }
 
+auto EntityBase::getDistanceToTargetLaneletPose(
+  const CanonicalizedLaneletPose & target_lanelet_pose) -> std::optional<double>
+{
+  constexpr double matching_distance = 100.0;  // may be better to use a parameter
+  const auto entity_lanelet_pose_ = getLaneletPose(matching_distance);
+  const auto target_lanelet_pose_ = target_lanelet_pose;
+
+  // check if the entity is on the lanelet
+  if (entity_lanelet_pose_) {
+    const auto entity_distance_to_target = hdmap_utils_ptr_->getLongitudinalDistance(
+      static_cast<LaneletPose>(entity_lanelet_pose_.value()),
+      static_cast<LaneletPose>(target_lanelet_pose_));
+
+    return entity_distance_to_target;
+  }
+
+  // may be give error here too?
+  return std::nullopt;
+}
+
+auto EntityBase::getIfArrivedToTargetLaneletPose(
+  const CanonicalizedLaneletPose & target_lanelet_pose, const double threshold) -> bool
+{
+  if (!getDistanceToTargetLaneletPose(target_lanelet_pose).has_value()) {
+    // error here
+  }
+  const auto entity_lanelet_pose_ = getLaneletPose();
+  const auto target_lanelet_pose_ = target_lanelet_pose;
+
+  const auto entity_distance_to_target = hdmap_utils_ptr_->getLongitudinalDistance(
+    static_cast<LaneletPose>(entity_lanelet_pose_.value()),
+    static_cast<LaneletPose>(target_lanelet_pose_));
+
+  if (entity_distance_to_target <= threshold) {
+    return true;
+  }
+  return false;
+}
+
+/** WARNING***************
+ * This should not be called from ego entity
+ */
```

Comment:
If you leave meme, please use doxygen.
And, please use THROW_SIMULATION_ERROR in EgoEntity.